### ISSUE TITLE
Crypto code review 

### ISSUE THREAD

**Issue by vkanne-msft** [mitls/mitls-fstar#80]

**Comment by markulf** [comment]
A first step is to define a process for reviewing our code:

- [ ] test the verification, by writing adversarial code, malicious interactions should result in typing errors
- [ ] write down cryptographic assumptions on paper

**Comment by markulf** [comment]
@vkanne-msft this is really more of a meta-issue or epic that should be split into separate tasks. For all code we write there should be some testing and review, especially for crypto code.

**Comment by markulf** [comment]
As we are developing an idealization style some relevant issues are discussed on the F* side, see https://github.com/FStarLang/FStar/issues/674

**Comment by markulf** [comment]
The push on the record paper definitely helped our understanding os some modules. Ticked in the above list. Still lots of work to be done, on verification and review.

**Comment by vkanne-msft** [comment]
@markulf Should we move this to Milestone 2

**Comment by markulf** [comment]
Yes, for the complete review. We should carve out a task for those modules that we verify for Milestone 1. Mostly on the Record protocol.
